File: 563. Binary Tree Tilt/solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
  int totalTilt = 0;
  
  int dfs(TreeNode current){
    if (current == null){
      return 0;
    }
    int left = dfs(current.left);
    int right = dfs(current.right);
    totalTilt += Math.abs(left - right);
    return current.val + left + right;
  }
  public int findTilt(TreeNode root) {
    dfs(root);
    return totalTilt;
  }
}

File: 563. Binary Tree Tilt/solution.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
  def findTilt(self, root):
    """
    :type root: TreeNode
    :rtype: int
    """
    self.totalTilt = 0
    
    def dfs(current):
      if current == None:
        return 0
      left = dfs(current.left)
      right = dfs(current.right)
      self.totalTilt += abs(left - right)
      return current.val + left + right
    
    dfs(root)
    return self.totalTilt